core/src/main/java/org/apache/brooklyn/core/mgmt/rebind/BasicEntityRebindSupport.java

```diff
@@ -245,14 +249,52 @@ protected void instanceRebind(AbstractBrooklynObject instance) {
         Preconditions.checkState(instance == entity, "Expected %s and %s to match, but different objects", instance, entity);
         Lifecycle expectedState = ServiceStateLogic.getExpectedState(entity);
         if (expectedState == Lifecycle.STARTING || expectedState == Lifecycle.STOPPING) {
-            LOG.warn("Entity {} goes on-fire because it was in state {} on rebind", entity, expectedState);
-            LOG.warn("not-up-indicators={}", entity.getAttribute(Attributes.SERVICE_NOT_UP_INDICATORS));
-            ServiceStateLogic.setExpectedState(entity, Lifecycle.ON_FIRE);
+            // If we were previously "starting" or "stopping", then those tasks will have been 
+            // aborted. We don't want to continue showing that state (e.g. the web-console would
+            // then show the it as in-progress with the "spinning" icon).
+            // Therefore we set the entity as on-fire, and add the indicator that says why.
+            markTransitioningEntityOnFireOnRebind((EntityInternal) entity, expectedState);
+        }
+        
+        // Clear the provisioning/termination task-state; the task will have been aborted, so wrong to keep this state.
+        ((EntityInternal)entity).sensors().remove(AttributesInternal.INTERNAL_PROVISIONING_TASK_STATE);
+        ((EntityInternal)entity).sensors().remove(AttributesInternal.INTERNAL_TERMINATION_TASK_STATE);
+        
+        super.instanceRebind(instance);
+    }
+    
+    protected void markTransitioningEntityOnFireOnRebind(EntityInternal entity, Lifecycle expectedState) {
+        LOG.warn("Entity {} being marked as on-fire because it was in state {} on rebind; indicators={}", new Object[] {entity, expectedState, entity.getAttribute(Attributes.SERVICE_NOT_UP_INDICATORS)});
+        ServiceStateLogic.setExpectedState(entity, Lifecycle.ON_FIRE);
+        ServiceStateLogic.ServiceNotUpLogic.updateNotUpIndicator(
+                entity, 
+                "Task aborted on rebind", 
+                "Set to on-fire (from previous expected state "+expectedState+") because tasks aborted on rebind");
```

Comment:
I've changed wording in the message.

---
I also looked again at how we set `INTERNAL_PROVISIONING_TASK_STATE`. In `org.apache.brooklyn.entity.software.base.lifecycle.MachineLifecycleEffectorTasks.ProvisionMachineTask`, there is a try-finally block, where it sets the sensor to "done" in the finally block. If that managed to execute + get persisted, then on rebind we wouldn't know that it was still executing!

We get away with this because the `managementContext.terminate()` stops the `rebindManager` and `BrooklynStorage`, before it shuts down the execution manager. Therefore our thread doesn't get interrupted (triggering the finally block) until after persistence has been disabled.